marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/QueryBatcherImpl.java

```diff
@@ -610,7 +608,7 @@ private synchronized void startQuerying() {
     public void run() {
       // don't proceed if this forest is marked as done (because we already got the last batch)
       AtomicBoolean isDone = forestIsDone.get(forest);
-      if ( isDone.get() == true ) {
+      if ( isDone.get() == true || jobDone.get()) {
```

Comment:
I notice the log message isn't appropriate for the jobDone case.  Maybe the test for jobDone should be a separate case instead of an OR?

Reply:
Yes. That's right.